src/modules/sub-items/services/bulk.service.js

```diff
@@ -0,0 +1,77 @@
+import { handleRequestResponse } from '../../common/helpers/request.helper.js';
+
+const HEADERS_BULK = {
+    Accept: 'application/vnd.ez.api.BulkOperationResponse+json',
+    'Content-Type': 'application/vnd.ez.api.BulkOperation+json',
+};
+
+export const bulkMoveLocations = (restInfo, locations, newLocationHref, callback) => {
+    const requestBody = getBulkMoveRequestOperations(locations, newLocationHref);
+
+    makeBulkRequest(restInfo, requestBody, processBulkResponse.bind(null, locations, 201, callback));
+};
+
+export const bulkDeleteContents = (restInfo, locations, callback) => {
+    bulkMoveLocations(restInfo, locations, '/api/ezp/v2/content/trash', callback);
+};
+
+const getBulkMoveRequestOperations = (locations, destination) => {
+    const operations = {};
+
+    locations.forEach((location) => {
+        operations[location.id] = {
+            uri: location._href,
+            method: 'MOVE',
+            headers: {
+                Destination: destination,
+            },
+        };
+    });
+
+    return operations;
+};
+
+const processBulkResponse = (locations, successCode, callback, response) => {
+    const { operations } = response.BulkOperationResponse;
+    const contentsMatches = { success: [], fail: [] };
+
+    for (const locationId of Object.keys(operations)) {
+        const response = operations[locationId];
+        const respectiveItem = locations.find((location) => String(location.id) === locationId);
+        const isSuccess = response.statusCode === successCode;
+
+        if (isSuccess) {
+            contentsMatches.success.push(respectiveItem);
+        } else {
+            contentsMatches.fail.push(respectiveItem);
+        }
+    }
+
+    callback(contentsMatches.success, contentsMatches.fail);
+};
+
+const makeBulkRequest = ({ token }, operations, callback) => {
+    const request = new Request(Routing.generate('ezplatform.bulk_operation'), {
+        method: 'POST',
+        headers: {
+            ...HEADERS_BULK,
+            'X-CSRF-Token': token,
+        },
+        body: JSON.stringify({
+            bulkOperations: {
+                operations,
+            },
+        }),
+        mode: 'cors',
```

Comment:
We use `'same-origin'`.

Reply:
I've taken it from: https://github.com/ezsystems/ezplatform-admin-ui-modules/blob/master/src/modules/sub-items/services/sub.items.service.js
It might be good when switching to something to change it everywhere, or at least write it down somewhere, otherwise it's hard to keep track of changes.